server/protocols/protocols-lmtp/src/main/java/org/apache/james/lmtpserver/hook/MailboxDeliverToRecipientHandler.java

```diff
@@ -80,15 +82,25 @@ public HookResult deliver(SMTPSession session, MailAddress recipient, MailEnvelo
             mailboxManager.getMailbox(MailboxPath.inbox(username), mailboxSession)
                 .appendMessage(MessageManager.AppendCommand.builder()
                     .recent()
-                    .build(envelope.getMessageInputStream()),
+                    .build(new Content() {

```

Comment:
I didn't go back to check all `Content` instances but it looks to be a lot of inline classes definitions that are exactly the same

Reply:
a lot of inline classes definitions that looks exactly the same. Nuance. I generalized where possible.